src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Lambda/LambdaEventHelpers.cs

```diff
@@ -183,18 +184,23 @@ private static void TryParseSQSDistributedTraceHeaders(dynamic sqsEvent, ITransa
                 }
             }
         }
-        else if (record.Body != null && record.Body.Contains("\"Type\" : \"Notification\"") && record.Body.Contains("\"MessageAttributes\""))
+        else if (record.Body != null && record.Body.Contains("\"Notification\"") && record.Body.Contains("\"MessageAttributes\""))
         {
             // This is an SNS subscription with attributes
-            foreach (var tracingKey in TracingKeys)
+            try
             {
-                var headerValue = TryParseAttributeFromSqsMessageBody(record.Body, tracingKey);
-                if (headerValue != null)
+                var snsMessage = WrapperHelpers.DeserializeObject<SnsMessage>((string)record.Body);
+                foreach (var messageAttribute in snsMessage.MessageAttributes)
                 {
-                    sqsHeaders.Add(tracingKey, headerValue);
+                    sqsHeaders.Add(messageAttribute.Key, messageAttribute.Value.Value);
                 }
             }
+            catch
+            {
+                // do nothing
```

Comment:
Might want to log the exception at Debug level just so we can see what the problem was - might also include `record.Body` in the message, as it'll help diagnosing the failure.

Reply:
Good idea, done.